include/nbl/builtin/glsl/utils/swapchain_transform.glsl

```diff
@@ -0,0 +1,119 @@
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR:
+        return ivec2(lastTexel.x - coord.x, coord.y);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_90:
+        return coord.yx;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_180:
+        return ivec2(coord.x, lastTexel.y - coord.y);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_270:
+        return lastTexel - coord.yx;
+    default:
+        return ivec2(0);
+    }
+}
+
+ivec2 nbl_glsl_swapchain_transform_backward(in uint swapchainTransform, in ivec2 coord, in ivec2 lastTexel) {
+    uint reverseTransform;
+    switch (swapchainTransform) 
+    {
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_90:
+        reverseTransform = NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_270;
+        break;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_270:
+        reverseTransform = NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_90;
+        break;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_90:
+        reverseTransform = NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_270;
+        break;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_270:
+        reverseTransform = NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_90;
+        break;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR:
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_180:
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_180:
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_IDENTITY:
+        reverseTransform = swapchainTransform;
+        break;
+    default:
+        return ivec2(0);
+    }
+    return nbl_glsl_swapchain_transform_forward(reverseTransform, coord, lastTexel);
+}
+
+ivec2 nbl_glsl_swapchain_transform_preTransform(in uint swapchainTransform, in ivec2 coord, in ivec2 screenSize) {
+    return nbl_glsl_swapchain_transform_backward(swapchainTransform, coord, screenSize - 1);
+}
+
+ivec2 nbl_glsl_swapchain_transform_postTransform(in uint swapchainTransform, in ivec2 coord, in ivec2 screenSize) {
+    return nbl_glsl_swapchain_transform_forward(swapchainTransform, coord, screenSize - 1);
+}
+
+ivec2 nbl_glsl_swapchain_transform_preTransformExtents(in uint swapchainTransform, in ivec2 screenSize) {
+    switch (swapchainTransform) 
+    {
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_IDENTITY:
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR:
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_180:
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_180:
+        return screenSize;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_90:
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_270:
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_90:
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_270:
+        return screenSize.yx;
+    default:
+        return ivec2(0);
+    }
+}
+
+mat2 nbl_glsl_swapchain_transform_postTransformMatrix(in uint swapchainTransform) {
```

Comment:
dont return a matrix, do the matrix mul inside (optimize it out) take `vec2` as parameter and return a `vec2`